contracts/v0.8/cbor/MarketCbor.sol

```diff
@@ -130,6 +130,10 @@ library MarketCBOR {
         buf.startFixedArray(uint64(params.deals.length));
 
         for (uint64 i = 0; i < params.deals.length; i++) {
+            bool isLabelStr = bytes(params.deals[i].proposal.label.dataStr).length > 0;
+            bool isLabelBts = params.deals[i].proposal.label.dataBts.length > 0;
+            require((!isLabelStr && isLabelBts) || (!isLabelBts && isLabelStr), "deal label must be either string or bytes");
```

Comment:
Not super deep into this code, but can't this be simplified?

```suggestion
            require(!(isLabelStr && isLabelBts), "deal label must be either string or bytes");
```

Also worth noting that this will default to serializing empty bytes if none is set, which I think is fine, but please document that behavior.

Reply:
Yes Raul, you are right! good catch